qcore/validate_vm.py

```diff
@@ -15,219 +15,238 @@
-            )
-        )
-        assert vm_params_dict[VMParams.nz.value] == int(
-            round(
-                (
-                    vm_params_dict[VMParams.extent_zmax.value]
-                    - vm_params_dict[VMParams.extent_zmin.value]
-                )
-                / vm_params_dict[VMParams.hh.value]
-            )
-        )
-    except AssertionError:
-        return (
-            False,
-            "VM config missmatch between extents and nx, ny, nz: {}".format(
-                vm_params_file_path
+    def get_expected_size(length):
+        return int(round(length / vm_params_dict[VMParams.hh.value]))
+
+    xyz_values = [
+        (
+            VMParams.nx.value,
+            vm_params_dict[VMParams.nx.value],
+            get_expected_size(vm_params_dict[VMParams.extent_x.value]),
+        ),
+        (
+            VMParams.ny.value,
+            vm_params_dict[VMParams.ny.value],
+            get_expected_size(vm_params_dict[VMParams.extent_y.value]),
+        ),
+        (
+            VMParams.nz.value,
+            vm_params_dict[VMParams.nz.value],
+            get_expected_size(
+                vm_params_dict[VMParams.extent_zmax.value]
+                - vm_params_dict[VMParams.extent_zmin.value]
             ),
-        )
+        ),
+    ]
+    errors = []
+    for name, actual, expected in xyz_values:
+        if actual != expected:
+            errors.append(
+                f"Dimension {name} had a value mismatch. Grid size {actual} does not match up with calculated size: {expected}"
+            )
+
+    if errors:
+        return False, "\n".join(errors)
+    return True, ""
+
+
+def validate_vm_files(vm_dir, srf=None):
+    """
+    Go through rules of VM directories. Return False if invalid.
+    vm_dir: folder path containing VM files
+    verbose: print progress
+    errors: print errors and warnings to stderr
+    """
+
+    vm_dir = Path(vm_dir)
+
+    errors = []
+    vel_crns_file = vm_dir / "VeloModCorners.txt"
+    vm_params_file_path = vm_dir / VM_PARAMS_FILE_NAME
+
+    # 1: has to exist
+    if not os.path.isdir(vm_dir):
+        return False, "VM dir is not a directory: {}".format(vm_dir)
+    # 2, 3: metadata files exist (made by gen_cords.py)
+    vm_params_dict = load_yaml(vm_params_file_path)
 
-    # 5: binary file sizes
+    vm_files = [vm_dir / "vs3dfile.s", vm_dir / "vp3dfile.p", vm_dir / "rho3dfile.d"]
+    all_files = [
+        *vm_files,
+        vel_crns_file,
+        vm_params_file_path,
+        vm_dir / f"gridout{vm_params_dict['sufx']}",
+        vm_dir / f"gridfile{vm_params_dict['sufx']}",
+        vm_dir / f"model_bounds{vm_params_dict['sufx']}",
+        vm_dir / f"model_coords{vm_params_dict['sufx']}",
+        vm_dir / f"model_params{vm_params_dict['sufx']}",
+    ]
+    for vm_file in all_files:
+        if not vm_file.exists():
+            errors.append(f"VM file not found: {vm_file}")
+
+    # 5, 6: binary file sizes
```

Comment:
Some of the numbering seems off in this file?

Reply:
Were you just keeping original numbering? It might be better to re-number the checks?

Reply:
Yeah, this is based off the original numbering. I should probably drop/re number it though